.ci/README.md

```diff
@@ -1,63 +1,87 @@
 Concourse CI tools for MagicModules and Google Providers
 ===
 
-These tools manage the downstream repositories of [magic-modules](https://github.com/GoogleCloudPlatform/magic-modules).
+These tools manage the downstream repositories of [magic-modules](https://github.com/GoogleCloudPlatform/magic-modules), and are collectively referred to as "The Magician".
 
-# Jobs
-The concourse pipeline defined here runs through four stages when a Github pull request is opened against `magic-modules`.
-## `mm-generate`
-The first job's goal is to generate the downstream repositories of MagicModules.  This is done almost entirely by the `generate` task.
+# CI For Downstream Developers
+If you're interested in developing the repositories that manage MagicModules, here are the things you'll want to know.
 
-### `generate`
-`generate` takes two inputs:
-* The magic-modules repository, with the pull request's `head` checked out and the repo in "detached `HEAD`" state.
-* This CI repository.
+## What The Magician Does
+The Magician takes the PR you write against MagicModules and creates the downstream (generated) commits that are a result of your changes.  It pushes the downstream commits to the relevant downstream repositories, and opens PRs against them, referencing your open PR against MagicModules.  When those PRs are merged, it updates the submodules under `build/` and merges your MagicModules PR.
 
-It then runs `magic-modules/generate.yml`, which specifies one output:
+## Your Workflow
 
-* The magic-modules repository, after generation has been accomplished.
+You'll write some commits, likely modifying files under `provider` or `products`, and push a PR.  Since The Magician needs to be able to add commits to your branch to update submodules, **make sure the branch of your PR is in `GoogleCloudPlatform/magic-modules`**.
```

Comment:
Maybe to make it even more explicit, add `(not your personal fork)` at the end

Reply:
Done.